net-misc/youtube-viewer/youtube-viewer-3.7.2.ebuild

```diff
@@ -0,0 +1,87 @@
+# Copyright 1999-2020 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit desktop eutils perl-module xdg-utils
+
+DESCRIPTION="A command line utility for viewing youtube-videos in Mplayer"
+HOMEPAGE="https://trizenx.blogspot.com/2012/03/gtk-youtube-viewer.html"
+SRC_URI="https://github.com/trizen/youtube-viewer/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="|| ( Artistic GPL-1+ )"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE="gtk2 gtk3"
+
+GTK_COMMON_DEPS="
+	dev-perl/File-ShareDir
+	virtual/freedesktop-icon-theme
+	x11-libs/gdk-pixbuf:2[X,jpeg]
+"
+RDEPEND="
+	dev-perl/Data-Dump
+	dev-perl/JSON
+	dev-perl/libwww-perl[ssl]
+	dev-perl/Term-ReadLine-Gnu
+	dev-perl/LWP-Protocol-https
+	virtual/perl-Encode
+	virtual/perl-File-Path
+	virtual/perl-File-Spec
+	virtual/perl-Getopt-Long
+	virtual/perl-Scalar-List-Utils
+	virtual/perl-Term-ANSIColor
+	virtual/perl-Term-ReadLine
+	virtual/perl-Text-ParseWords
+	virtual/perl-Text-Tabs+Wrap
+	gtk2? (
+		>=dev-perl/Gtk2-1.244.0
+		${GTK_COMMON_DEPS}
+	)
+	gtk3? (
+		dev-perl/Gtk3
+		${GTK_COMMON_DEPS}
+	)
+	|| ( media-video/ffmpeg[openssl,-libressl] media-video/ffmpeg[-openssl,libressl] media-video/ffmpeg[gnutls] )
+	|| ( media-video/mpv media-video/mplayer media-video/vlc media-video/smplayer )"
+DEPEND="dev-perl/Module-Build"
+
+src_configure() {
+	local myconf
+	if use gtk2 && use gtk3; then
+		myconf="--gtk"
+	elif use gtk2; then
+		myconf="--gtk2"
+	elif use gtk3; then
+		myconf="--gtk3"
```

Comment:
I don't think this logic is how you want it to be. You should always enable gtk3 if it's used. Maybe you want to introduce some REQUIRED_USE constraint there?

Reply:
So no gtk2 only build option essentially?

Reply:
I mean, `USE="gtk2 gtk3"` should pick gtk3 rather than gtk1. You could make a REQUIRED_USE constraint to pick at-most-one-of when choosing a GTK engine. I'd deprecate gtk1 altogether.

Reply:
The flags are misleading here, there is no gtk1 support. The --gtk configuration flag from upstream will install both gtk2 and gtk3 UIs. So currently the user can choose to have:
- no UI (only CLI mode) by setting none of the use flags
- gtk2 only interface (via --gtk2 build config switch) by setting gtk2 use flag
- gtk3 only interface (via --gkt3 build config switch) by setting gtk3 use flag
- both gtk2 and gtk3 interfaces (via --gtk build config switch) by setting both gtk2 and gtk3 use flags.

I could do a REQUIRED_USE to make gtk2 and gtk3 interfaces mutually exclusive however it raises the question what is more desirable:
- let the users choose what they want (i.e. multi user machine where different users prefer different UIs)
- force only one of the flags as we are wasting disk space if both gtk2 and gtk3 flags are set from global USE but user really only uses whatever comes up as default.

Also I'm not sure about gtk2's status? Is it being deprecated in favour of gtk3 in general?

Reply:
Okay, I think it's fine then. Gtk2 has been on deprecation for a long time.. but it's not going away soon. So let's not add REQUIRED_USE For now. I'll check this PR next when I resume proxy-maint work (probably in tomorrow morning)

Reply:
I've decided to follow the Gnome project guideline so the choices are now:
- no GUI if no gtk flag is set
- USE="gtk" will install gtk3 gui by default
- USE="gtk gkt2" will install GUI, but downgrade it to gkt2 as per the user's request.

I also added REQUIRED_USE for this.

I think once the gkt3 UI gets stable in future versions I will drop gtk2 support completely.